### ISSUE TITLE
Can't Drag/Move Connections into Folders

### ISSUE THREAD

**Issue by pgalbraith** [dbeaver/dbeaver#18799]
### Description

1. Create connection
2. Create folder
3. Try to drag connection into folder using mouse
4. Fail

A pretty easy workaround is to copy/paste the connection into the desired folder, and then delete the original.

### DBeaver Version

CE 22.3.2

### Operating System

Ubuntu 22.04 w/X.org

**Comment by LonwoLonwo** [comment]
Hello @pgalbraith 

What OS do you have?

**Comment by LonwoLonwo** [comment]
related? #17415

**Comment by pgalbraith** [comment]
Ah yeah that looks like it's my problem.  I'm running on Ubuntu using x.org (wayland disabled).

**Comment by pgalbraith** [comment]
Likely duplicate of #17415

**Comment by LonwoLonwo** [comment]
Sorry, I do not understand. Is your problem solved? Did you try to update DBeaver? Because https://github.com/dbeaver/dbeaver/issues/17415 is closed already.

**Comment by pgalbraith** [comment]
@LonwoLonwo I saw there was a commit in December re: #17415 so I was just assuming this hadn't made it to a release yet, but I'll re-open this just in case.

**Comment by KarimTayie** [comment]
Same issue here. same operating system.

**Comment by mcdoogs** [comment]
Same issue here, but I'm on Arch. Kernel 6.1.8/DE: Plasma 5.26.5/x11.
I noticed the issue on dbeaver version 22.3.2, so I downloaded 22.3.3 to test if it's fixed, but the behavior is the same. I know I've drag-n-dropped successfully at some point in the past, but not sure how long ago.

The issue is the same both in the 'database navigator' and 'projects' view. Drag and drop does not work for moving connections between folders, moving folders into other folders, rearranging folders or connections

https://user-images.githubusercontent.com/49351525/216105254-bf889b7e-87e8-4319-9761-ac8a46cf28b9.mp4

Edit: I can also confirm that switching to a Wayland session fixes the issue, so this is x.org/x11 specific.

**Comment by flavio-evocorp** [comment]
I have the same problem!
I'm used distro Ubuntu 20.04.2 LTS

**Comment by gustavotrott** [comment]
It is happening for me using Ubuntu 22.04 with Xorg.

_As workaround I copied the connections and pasted inside the folder.
And then deleted the old ones._

**Comment by flavio-evocorp** [comment]
Update ubuntu for 22.04.2 LTS and dbeaver  23.0.0.
It's working!

**Comment by gustavotrott** [comment]
It's not working here!

![test](https://user-images.githubusercontent.com/5660191/224853978-86946ab2-c834-4ada-af6b-453b4a55c2f6.gif)

![dbeaver-version](https://user-images.githubusercontent.com/5660191/224853963-3b4d47fb-ade3-4f1f-9a72-481ca39242d2.png)

![image](https://user-images.githubusercontent.com/5660191/224854120-f615da2e-797b-4f1e-ae83-25bfe1437e83.png)

**Comment by flavio-evocorp** [comment]
It's strange, I'm with this exact versions on ubuntu and dbeaver e it's working! 

![Dbeaver](https://cdn.discordapp.com/attachments/872489839818322010/1085159161458851900/image.png)

![Ubuntu](https://media.discordapp.net/attachments/872489839818322010/1085159339926507530/image.png)

**Comment by mcdoogs** [comment]
> It's strange, I'm with this exact versions on ubuntu and dbeaver e it's working! 
> 
> ![Dbeaver](https://cdn.discordapp.com/attachments/872489839818322010/1085159161458851900/image.png)
> 
> ![Ubuntu](https://media.discordapp.net/attachments/872489839818322010/1085159339926507530/image.png)
> 
> 
> 

It's because you're using Wayland now, the upgrade to Ubuntu must have changed the default for you.

**Comment by flavio-evocorp** [comment]
You are right! Before was "X11". Thanks by informations!

**Comment by thiagobraga** [comment]
I'm also having this issue.
I'm using Ubuntu 22.04 with Xorg.
DBeaver 23.0.3.202304231635

**Comment by ikodev** [comment]
Same issue here T_T
DBeaver Version 23.0.3.202304240616
Ubuntu 22.04.2 LTS

**Comment by oraclejavanet** [comment]
Not working for me. Same issue: Cannot drag/drop connections into folders.
Rocky Linux release 8.7 (Green Obsidian)
DBeaver Version 23.1.2.202307091549

**Comment by Conzar** [comment]
I can confirm this issue does not work for me either:
**Dbeaver**: Version 23.1.4.202308061552
**OS**: Ubuntu 22.04
**Display Server**: X11

## Work around
* Copy and Paste the connection into the folder
* Delete the old connection
* Rename the new connection (removing the '1' at the end of the name)

**Comment by DDHF22** [comment]
Does the workaround preserve the "SQL Editor"->"Recent SQL script" ?
I use that all the time on 50+ connections

Update: Think the answer is no :-( ( DBeaver 23.2.1.202309250303 @ Ubuntu 22.04.3 LTS)

**Comment by DDHF22** [comment]
Guess you can tweak some files in
~/snap/dbeaver-ce/current/.local/share/DBeaverData/workspace6/General/.dbeaver
but I would prefer not to do that as I may miss some references

**Comment by DDHF22** [comment]
Correct workaround is doing right click "Edit Connection" and modify folder to any predefined folder in the dropdown field.

![image](https://github.com/dbeaver/dbeaver/assets/1453308/49a5e4a4-eae2-480e-be07-352fc92a17d9)

**Comment by fpontef** [comment]
> Does the workaround preserve the "SQL Editor"->"Recent SQL script" ?

Thanks for the warning, I didn't know this would happen after the copy procedure.

I have Dbeaver installed on Arch and following this path I could get my scripts back.
`~/.local/share/DBeaverData/workspace6/General/Scripts/`

Using "snap" I think you can follow the path to copy your .sql scripts: `~/snap/dbeaver-ce/current/.local/share/DBeaverData/workspace6/General/Scripts/`

**Comment by mzatko** [comment]
same problem here, ubuntu 23.10, dbeaver 23.2.5

**Comment by roma2341** [comment]
Not workin
ubuntu 22.04
DBeaver 23.3.5.202402181738

**Comment by curtisbelt** [comment]
Still occurring on DBeaver 24.0.0.202403031740

- **OS Name:**                                     Fedora Linux 39 (Workstation Edition)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.5
- **Windowing System:**                            X11
- **Kernel Version:**                              Linux 6.7.9-200.fc39.x86_64

In the meantime, as [posted above](https://github.com/dbeaver/dbeaver/issues/18799#issuecomment-1737693581), it works fine to use `Right click connection -> Edit Connection -> General tab -> Connection folder`.

**Comment by TiS** [comment]
Yeah, for connection the workaround works fine. But if you want to move **folder**, you don't have such option.

BTW - it still does not work for me

* **OS Name:** Ubuntu 23.10
* **Windowing system:** X11
* **DBeaver Version:** 24.1.0.202406021658

**Comment by ArmoredPony** [comment]
Same issue. Linux Mint 21.3, X11, DBeaver 24.2.1

**Comment by Cidro** [comment]
Same issue here, Ubuntu 24.04 Using Gnome and X11. 
I can't move folders or connection into another

**Comment by MLengl** [comment]
Same issue on using:
- Mint 22 Cinnamon (6.8.0-48-generic)
- DBeaver Community Version 24.2.4.202411031831

Because Drag&Drop is particially laggy or affected by isolation siwtch snap/flatpak I have tested the .DEB and flatpack. Also flatpark with changing home to my real and not sandbox home.
All of this installation/Setup version had the same behaviour - drag&move works but it will not stay in the folder on dropping.

Applies on "Database Navigator" and "Projects" UI.
So they are behaving the same which, in my opinion, indicates it is a general and not OS based or View based issue.

Last notes/workarounds:
- Copy and paste on folder is working
- Manual shift with "Edit connection > General > Connection Folder" works too.

**Comment by StevenBrons** [comment]
Same issue on Linux Mint 21.2

**Comment by gustavotrott** [comment]
I upgraded Ubuntu and Dbeaver, and now I can no longer reproduce!!

<table><tr>

<td>

![Image](https://github.com/user-attachments/assets/601b317b-4bda-4c39-af77-4e1d228c5ee0)

</td>

<td>

![Image](https://github.com/user-attachments/assets/f52622e8-98be-4499-9a1d-be85167270c0)

</td>



</tr></table>

**Comment by ShadelessFox** [comment]
Caused by #17415
